sw/device/examples/sram_program/BUILD

```diff
@@ -128,12 +126,60 @@ opentitan_gdb_fpga_cw310_test(
         print sram_main()
 
         echo :::: Done.\\n
-    """,
-    gdb_script_symlinks = {
-        ":sram_program_fpga_cw310.elf": "sram_program.elf",
-        "//sw/device/silicon_creator/rom:rom_fpga_cw310.elf": "rom.elf",
-    },
-    rom_bitstream = "//hw/bitstream:rom_otp_exec_disabled",
-    rom_kind = "Rom",
-    tags = ["manual"],
```

Comment:
Just saw CI failures in the `sw_test` job because `updatemem` is not installed. Maybe I should keep this "manual" tag.

Reply:
But then, I suppose the tests won't be grabbed by [run-fpga-cw310-tests.sh](https://cs.opensource.google/opentitan/opentitan/+/master:ci/scripts/run-fpga-cw310-tests.sh;l=41;drc=46b156a66e1d0e4f7e651b55790347a8fae4fbc0). Hmm.

Reply:
@jprendes I think update mem is available in the fpga runners but not in azure runners, right? Also, it should be in path, i.e. we should run settings64.sh unless it's already run.

@dmcardle can you check if this failure is happening on azure? we should probably run these only on fpga runners.

Reply:
Yeah, it's not surprising that updatemem is unavailable on `sw_build` because it's just running on a standard Ubuntu image. I think I just forgot to add the "cw310" tag to the new `test_suite`, so it didn't get [filtered out](https://cs.opensource.google/opentitan/opentitan/+/master:azure-pipelines.yml;l=218;drc=3b34c6f21cafc64bfa6f2d5fe768d1e8a0d01215).

Reply:
(Already added the tag a minute ago, waiting for CI to confirm my hypothesis.)

Reply:
I was wrong, it happened again on `sw_build`: https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=98326&view=logs&j=5df5adf8-3577-5927-f0b2-31afed2697ee&t=1253ba27-4e8c-57af-d0ba-888237644875&l=4195

```
(14:44:26) ERROR: /home/vsts/work/1/s/sw/device/examples/sram_program/BUILD:153:21: SpliceBitstream sw/device/examples/sram_program/rom_otp_test_unlocked0_exec_disabled.spliced.bit failed: (Exit 1): updatemem failed: error executing command 
  (cd /home/vsts/.cache/bazel/_bazel_vsts/5c27df7f718ab35828b0e0cfe0993663/execroot/lowrisc_opentitan && \
  exec env - \
    HOME=/home/vsts \
    PATH=/home/vsts/.cache/bazelisk/downloads/bazelbuild/bazel-5.1.1-linux-x86_64/bin:/home/vsts/.cargo/bin:/tools/verible/bin:/home/vsts/.local/bin:/home/vsts/.local/bin:/opt/pipx_bin:/home/vsts/.cargo/bin:/home/vsts/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/vsts/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin \
    XILINXD_LICENSE_FILE='' \
    XILINX_HLS='' \
    XILINX_VIVADO='' \
  updatemem -force --meminfo external/bitstreams/cache/7c5837b7ada5c32280c9c18b473388354766f807/otp.mmi --data bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/examples/sram_program/rom_otp_test_unlocked0_exec_disabled.update.mem --bit bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/examples/sram_program/rom_otp_test_unlocked0_exec_disabled.tmpsrc.bit --proc dummy --out bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/examples/sram_program/rom_otp_test_unlocked0_exec_disabled.spliced.bit --debug)
# Configuration: 912f1611c8883fc351b9588c07b294968bf463aa7b6df148f6170cca2368f6cd
# Execution platform: @local_config_platform//:host
src/main/tools/process-wrapper-legacy.cc:80: "execvp(updatemem, ...)": No such file or directory
```

I think this might have something to do with the `--build_tests_only=false` that we specify in `sw_build`. Stepping out for a bit, I'll have to pick this up later.

Reply:
> I think this might have something to do with the --build_tests_only=false that we specify in sw_build.

Expanding on that, I think that `--build_tests_only=false` is causing us to build the new tests, and thus the new `bitstream_splice` targets that fail because `updatemem` is not present. I could add `--build_tag_filters=-cw310`, but that would lose build coverage for everything else with that tag.

I think this would be solved neatly by caching all our bitstream artifacts in GCP and making use of the select-based targets in //hw/bitstream. But that would require us to fix #15284.

Perhaps in the meantime, I'll add a "vivado" tag that indicates the target requires the Vivado toolchain? Then we'd just need to add `--build_tag_filters=-vivado` to the `sw_build` job.

Reply:
@milesdai @drewmacrae @cfrantz Do you folks have thoughts on adding a "vivado" tag as described above?

Reply:
SGTM, I'll merge this PR now since the meaning of `-vivado` is pretty clear IMO. We can revisit this if we hear additional feedback before #15284 is addressed.

Reply:
A common problem with build_tag_filters is they need to be applied to any targets that depend on them as well. It might be better to exclude these with a query, or satisfy them with some sort of dummy as we do with verilator.

Reply:
Option 1: Tag every target that transitively depends on `bitstream_splice` with "vivado". Modify every Bazel invocation in azure-pipelines.yml to exclude "vivado".

Option 2: Change every Bazel invocation in azure-pipelines.yml to exclude targets that transitively depend on a `bitstream_splice` rule. No tagging required, but it makes CI a little harder to read.

Option 3: Tag every target that transitively depends on `bitstream_splice` with "manual" and "not_manual_if_vivado". Modify run-fpga-cw310-tests.sh to explicitly select tests with that label.

Thoughts on these options? I'm favoring option 3.

Reply:
If everything that isn't a test and transitively depends on bitstream_splice is manual, as described in Option 3, and everything that is a test is tagged with vivado, then test_tag_filters will behave mostly as people expect them, and the downside is we miss out on build coverage unless we consciously enable it on a system with vivado installed.

Reply:
I'd like to write some checks that tests that depend on verilator have a verilator tag, and checks that depend on vivado have a vivado tag.